argopy/fetchers.py

```diff
@@ -264,29 +312,161 @@ def to_xarray(self, **kwargs):
+                index_box = self._AccessPoint_data['box']
+                del index_box[4:6]
+                index_loader.region(index_box).load()
+            df = index_loader.index
+
+            if self._loaded and self._mode == 'standard' and len(self._index) != len(df):
+                warnings.warn("Loading a full index in 'standard' user mode may lead to more profiles in the "
+                              "index than reported in data.")
+
+            # Possibly replace the light index with the full version:
+            if not self._loaded or self._request == self.__repr__():
+                self._index = df
+
+        return df
+
+    def load(self, force: bool = False, **kwargs):
+        """ Load data in memory
+
+            Apply the default to_xarray() and to_index() methods and store results in memory.
+            Access loaded measurements structure with the `data` and `index` properties::
+
+                ds = ArgoDataFetcher().profile(6902746, 34).load().data
+                # or
+                df = ArgoDataFetcher().float(6902746).load().index
+
+            Parameters
+            ----------
+            force: bool
+                Force loading, default is False.
+
+            Returns
+            -------
+            :class:`argopy.fetchers.ArgoDataFetcher.float`
+                Data fetcher with `data` and `index` properties in memory
+        """
+        # Force to load data if the fetcher definition has changed
+        if self._loaded and self._request != self.__repr__():
+            force = True
+
+        if not self._loaded or force:
+            # Fetch measurements:
+            self._data = self.to_xarray(**kwargs)
+            # Next 2 lines must come before ._index because to_index() calls back on .load() to read .data
+            self._request = self.__repr__()  # Save definition of loaded data
+            self._loaded = True
+            # Extract measurements index from data:
+            self._index = self.to_index(full=False)
+        return self
 
     def clear_cache(self):
         """ Clear data cached by fetcher """
         if not self.fetcher:
-            raise InvalidFetcher(" Initialize an access point (%s) first." %
-                                 ",".join(self.Fetchers.keys()))
+            raise InvalidFetcher(
+                " Initialize an access point (%s) first."
+                % ",".join(self.Fetchers.keys())
+            )
         return self.fetcher.clear_cache()
 
+    def plot(self, ptype="trajectory", **kwargs):
+        """ Create custom plots from data
+
+            Parameters
+            ----------
+            ptype: str
+                Type of plot to generate. This can be: 'trajectory',' profiler', 'dac'.
+
+            Returns
+            -------
+            fig: :class:`matplotlib.figure.Figure`
+            ax: :class:`matplotlib.axes.Axes`
+        """
+        self.load()
+        if ptype in ["dac", "institution"]:
```

Comment:
So here we should check if our `self.index` dataframe is "full". If not, we must `self.to_index(full=True)` before we can bar_plot by dac or profiler

Reply:
Or we could tell the user to do so... to avoid perf issues with the erddap index dataset

Reply:
Yes. 
and also for the `profiler` plot ...